src/AutoMerge/guidelines.jl

```diff
@@ -25,7 +26,8 @@ const guideline_compat_for_julia = Guideline(;
         "There is an upper-bounded `[compat]` entry for `julia` that ",
         "only includes a finite number of breaking releases of Julia.",
     ),
-    check=data -> meets_compat_for_julia(data.registry_head, data.pkg, data.version),
+    check=(data, guideline_parameters) ->
```

Comment:
Why can't `guideline_parameters` be included in `data`?
Or well, it looks like there is new fields for them in `data`, so why is it also an extra argument for the checks?

Reply:
I meant to remove them from GitHubAutoMergeData once I decided it wasn't appropriate for that struct to accrete non-GutHub API related stuff.

GitHubAutoMergeData isn't created untill pull_requedt_build so guideline parameters would need to be passed to pull_reqyest_build.

Sorry if there are typos or misremembered identifiers.  I'm typing on my phone on a very bouncy bus ride.

Reply:
Linecount parameters have been removed from GitHubAutoMergeData.

Reply:
Most of the fields in the struct are not GitHub specific. Once someone implements support for some more backend, there are good reasons to refactorize the struct, but the real point of the struct is to collect the data needed by the guidelines so we don't have to pass around lots of arguments.

Reply:
There is no reason to not start that refactoring now.

The real point of the struct, if the name is to be believed, is to communicate with the GitHub API and provide it's results.

That it has been co-opted as a repository for more general parameters is unfortunate.  I was able to restrain myself from doing that further.

Like I said before, it is created too late to avoid having to declare arguments for each guideline parameter for pull_request_build.  I'm willing to migrate parameters from function arguments and GirHubAutoMergeData to the fict, but not in this PR.

Most of the work I've put in to this PR was in tracing how existing guideline parameters fliwed through the code and tracking how they might get modified during that journey.  I hope to make that much clearer for the next guy.